tests/testingcharm/src/charm.py

```diff
@@ -26,33 +24,21 @@
 class TestingcharmCharm(CharmBase):
     """Charm the service."""
 
-    _stored = StoredState()
-
     def __init__(self, *args):
         super().__init__(*args)
-        self._stored.set_default(pushgateway_url=None)
 
         self.framework.observe(self.on.install, self._on_install)
-        self.framework.observe(self.on.pushgateway_relation_created, self._on_push_relation)
-        self.framework.observe(self.on.pushgateway_relation_changed, self._on_push_relation)
         self.framework.observe(self.on.send_metric_action, self._on_send_metric)
 
+        self.ppi = PrometheusPushgatewayInterface(self)
```

Comment:
as above: I'd avoid the acronym

Reply:
but this is testing code, so yeah... who cares.

Reply:
Improved anyway.

Reply:
Improved.